ironfish/src/storage/database.test.ts

```diff
@@ -559,6 +560,43 @@ describe('Database', () => {
       await transaction.commit()
       expect(await db.metaStore.get('test')).toBe(2)
     })
+
+    it('should wait for a lock before executing the handler', async () => {
+      await db.open()
+
+      let value = null
+
+      const [waitingPromise, waitingResolve] = PromiseUtils.split<void>()
+
+      // Queue up two transactions
+      const t1 = db.transaction([db.metaStore], 'readwrite', async () => {
+        value = 't1'
+        await waitingPromise
+      })
+
+      const t2 = db.transaction([db.metaStore], 'readwrite', async () => {
+        value = 't2'
+        await waitingPromise
+      })
+
+      const t3 = db.transaction([db.metaStore], 'readwrite', async () => {
+        value = 't3'
+        await waitingPromise
+      })
+
+      await PromiseUtils.sleep(0)
```

Comment:
Wish we had a comment here, no one will be able to figure out why this is here because of how Lock works